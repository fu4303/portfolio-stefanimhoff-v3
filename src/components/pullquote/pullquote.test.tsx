import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { Pullquote } from '.';
import type { PullquoteProps } from '.';

type PullquoteTestProps = PullquoteProps & {
  description: string;
};

describe('Pullquote', () => {
  // prettier-ignore
  const combinations: PullquoteTestProps[] = [
    { description: 'default', children: 'Pullquote' },
    { description: 'german', children: 'Pullquote', lang: 'de' },
    { description: 'japanese', children: 'Pullquote', lang: 'ja' },
    { description: 'author', children: 'Pullquote', author: 'author' },
    { description: 'source', children: 'Pullquote', source: 'source' },
    { description: 'source, sourceUrl', children: 'Pullquote', source: 'source', sourceUrl: 'sourceUrl' },
    { description: 'author, source, sourceUrl', children: 'Pullquote', author: 'author', source: 'source', sourceUrl: 'sourceUrl' },
    { description: 'author, source', children: 'Pullquote', author: 'author', source: 'source' },
  ];

  combinations.forEach((combination) => {
    test(`renders in ${combination.description}`, () => {
      const { container } = render(<Pullquote {...combination} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  combinations.forEach((combination) => {
    test(`is accessible in ${combination.description}`, async () => {
      const { container } = render(<Pullquote {...combination} />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
