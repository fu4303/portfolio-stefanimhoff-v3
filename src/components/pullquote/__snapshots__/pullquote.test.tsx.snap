// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pullquote renders in author 1`] = `
.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  >
    <b
      class="c2"
    >
      author
    </b>
  </footer>
</blockquote>
`;

exports[`Pullquote renders in author, source 1`] = `
.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  >
    <b
      class="c2"
    >
      author
    </b>
    , 
    <cite
      class=""
    >
      source
    </cite>
  </footer>
</blockquote>
`;

exports[`Pullquote renders in author, source, sourceUrl 1`] = `
.c3 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

.c2 {
  font-style: normal;
}

<blockquote
  class="c0"
  lang="en"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  >
    <b
      class="c2"
    >
      author
    </b>
    , 
    <cite
      class=""
    >
      <a
        class="c3"
        href="sourceUrl"
        rel="nofollow noopener noreferrer external"
      >
        source
      </a>
    </cite>
  </footer>
</blockquote>
`;

exports[`Pullquote renders in default 1`] = `
.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

<blockquote
  class="c0"
  lang="en"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  />
</blockquote>
`;

exports[`Pullquote renders in german 1`] = `
.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

<blockquote
  class="c0"
  lang="de"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  />
</blockquote>
`;

exports[`Pullquote renders in japanese 1`] = `
.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

<blockquote
  class="c0"
  lang="ja"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  />
</blockquote>
`;

exports[`Pullquote renders in source 1`] = `
.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

<blockquote
  class="c0"
  lang="en"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  >
    <cite
      class=""
    >
      source
    </cite>
  </footer>
</blockquote>
`;

exports[`Pullquote renders in source, sourceUrl 1`] = `
.c2 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  padding: clamp( 1.2rem,2.57vw,1.44rem);
  text-align: center;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0 p {
  font-size: clamp(1.25rem,1.8vw,2.00rem);
  font-weight: 900;
  line-height: 1.0;
  margin: 0;
}

.c0 p:first-child::before {
  content: open-quote;
  font-style: normal;
}

.c0 p:nth-of-type(1):first-letter {
  margin-left: -0.4em;
}

.c0[lang|='de'] p:nth-of-type(1):first-letter {
  margin-left: -0.6em;
}

.c0[lang|='ja'] p:nth-of-type(1):first-letter {
  margin-left: -1em;
}

.c0 p:nth-last-of-type(1)::after {
  content: close-quote;
  font-style: normal;
}

.c1 {
  font-size: clamp(0.65rem,0.8vw,0.75rem);
  font-weight: normal;
  margin-top: clamp( 0.694rem,1.49vw,0.833rem);
}

<blockquote
  class="c0"
  lang="en"
>
  <p>
    Pullquote
  </p>
  <footer
    class="c1"
  >
    <cite
      class=""
    >
      <a
        class="c2"
        href="sourceUrl"
        rel="nofollow noopener noreferrer external"
      >
        source
      </a>
    </cite>
  </footer>
</blockquote>
`;
